### PR TITLE
Adds a programmer workflow to the Documentation

### PR DESCRIPTION
**Pull request by kordjamshidi**

### CODE REVIEW

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/TwitterSentimentAnalysis/twitterSentimentApp.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  // Set up language filters
+  //List<String> languages = Arrays.asList("en", "es");
+
+  val client: TwitterClient = new TwitterClient(null, locations, null);
+  ClassifierUtils.LoadClassifier(sentimentClassifier)
+  // A separate thread for handling the queue of tweets
+  val messageHandler: ClassifierMessageHandler = new ClassifierMessageHandler(client.getMsgQueue(), client.getClient(), sentimentClassifier.classifier);
+  val thread: Thread = new Thread(messageHandler);
+  thread.start();
+}
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/TwitterSentimentAnalysis/twitterSentimentApp.scala]: I am not sure from where this object comes, was n't there before!? @bhargav

Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/TwitterSentimentAnalysis/twitterSentimentApp.scala]: Oh, I see it is a name change here compared to my documentation branch, removing it ...

**Comment by kordjamshidi** [comment]
We need to work more on the documentation: I was surprised to see we did not have any information about how to call test and train over classifiers in the current documentation, so I dded some.

**Comment by kordjamshidi** [comment]
Can someone review this? should be quick. @bhargav @danyaljj

File: README.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ Visit each link for its content
 <a name="Tutorial"></a>
  1. [Introduction](saul-core/doc/INTRO.md)
  2. [Installation] (saul-core/doc/INSTALLATION.md)
+ 3. [Programmers workflow] (saul-core/doc/WORKFLOW.md)
  3. [Conceptual structure of a Saul program](saul-core/doc/CONCEPTUALSTRUCTURE.md)
```
Comment (danyaljj) [inline README.md]: Could you fix the numbering of the items? (like after 3 it should be 4, etc).

File: saul-core/doc/SAULLANGUAGE.md [hunk 3]
```diff
@@ -17,6 +17,18 @@ object OrgClassifier extends Learnable[ConllRawToken](ErDataModelExample) {
     containsInPersonList, wordLen, containsInCityList)
 }
 ```
+### train and test classifiers
+
+call `train()` method to train your classifier using the populated data in the data model's training instances:
```
Comment (danyaljj) [inline saul-core/doc/SAULLANGUAGE.md]: Start sentence with capital letter?

File: saul-core/doc/SAULLANGUAGE.md [hunk 4]
```diff
@@ -17,6 +17,18 @@ object OrgClassifier extends Learnable[ConllRawToken](ErDataModelExample) {
...
+
+```scala
+OrgClassifier.train()
+```
+call `test()` method to test your classifier using the populated data model's test instance:
+
+ ```scala
+ OrgClassifier.test()
+```
```
Comment (danyaljj) [inline saul-core/doc/SAULLANGUAGE.md]: Should we or we shouldn't explain the other possible arguments for the `test` function? 
https://github.com/IllinoisCogComp/saul/blob/master/saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/Learnable.scala#L289-L318

File: saul-core/doc/WORKFLOW.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: Maybe add a title for the page? 
```
# Saul Workflow 
```

File: saul-core/doc/WORKFLOW.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+A Saul programmer needs to think about doing this workflow.
+
+1) Prepare a Reader program can be written in Java or Scala to read the data in some class of objects.
```
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: drop "can be"?

Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: "Class" is confusing with Java-classes. Maybe paraphrase "in some class of objects", or drop it?

File: saul-core/doc/WORKFLOW.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+2) Create a datamodel object by opening a new file and input your declaration in this template ([see more](DATAMODELING.md)):
+
+ ```
```
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: scala style?

File: saul-core/doc/WORKFLOW.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+ object myClassifierName1 extends Learnable(myNode){the definition of features, label and algorithm comes here!}
+ ...
+```
+4) For the advanced usage of Saul when you deal with predicting structured output and using constraints you open another object and define
+your global constraints in there in the following template (for the classic machine learning usage you can skip this step): ([see more](SAULLANGUAGE.md))
+
+ ```scala
+ object allMyConstraints {
+     val myFirstConstraintName = ConstrainedClassifier.constraint[ObjectType] { x: ObjectType =>
+     myLogicalExpression that describes my constrain comes here!
```
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: make this a comment by adding a `//`? (just to be consistent with the snippets below)
Same for the snippet above.

**Comment by kordjamshidi** [comment]
Why semaphore is complaining?

**Comment by danyaljj** [comment]
It's not Semaphore. It's anther CI for testing on Windows OS. 
Almost surely has nothing to do with this PR. @bhargav will probably address it at some point.

**Comment by kordjamshidi** [comment]
ah, ok, so please merge then if this PR is ok.

**Comment by bhargav** [comment]
I was testing out CI for Windows. Enabled it on the main repo by mistake. Was testing it on my fork. Semaphore build passed.

File: saul-core/doc/WORKFLOW.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      x: ObjectType =>
+     //myLogicalExpression that describes my constrain comes here!
+   }
+```
+5) Having the above items in place, now you can write your application that loads the data (population of the data into the data model) trains over the data, tests the models
+makes predictions or whatever you need in your learning based program, this is a typical scenario for a main app that uses classifiers ([see more](CONCEPTUALSTRUCTURE.md)):
+
+```scala
+object myMainApplication extends App{
+// call your reader here and read the data in some collection
```
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: Sorry for being nitpicky; could you fix the indentation of the code? New lines inside brackets should have 2 extra space. 
Also some snippets start with some space (like first one that has 3 spaces), and some don't have any space before (like this one). Could you make them consistent?

File: saul-core/doc/WORKFLOW.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
+
+##Saul Workflow
+A Saul programmer needs to think about doing this workflow.
+
+1) Prepare a Reader program written in Java or Scala to read the data in some Java or Scala collections of objects.
+2) Create a datamodel object by opening a new file and input your declaration in this template ([see more](DATAMODELING.md)):
```
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: Add a new line between 1) and 2)? Otherwise 2) doesn't appear in a new line ...

**Comment by danyaljj** [comment]
Just two remaining comments (code indentation, and number of the items in the root readme), and then good to merge.

**Comment by kordjamshidi** [comment]
do you mean by code indentation here? thanks for the point, fixed the numbering.

File: saul-core/doc/WORKFLOW.md (hunk 9 above)
Comment (danyaljj) [inline saul-core/doc/WORKFLOW.md]: I was referring the code snippets on this file. They have different indentations.

**Comment by kordjamshidi** [comment]
Markdown preview looks weird in my Intellij and I can not see the correct spacing but try to fix.

**Pull request merged**